tests/integration/security/authorization_test.go

```diff
@@ -940,118 +1009,121 @@ func TestAuthorization_Conditions(t *testing.T) {
-								newTestCase(b[0], "/request-headers", map[string]string{"x-foo": "bar"}, false),
-								newTestCase(a[0], "/request-headers", nil, false),
-								newTestCase(b[0], "/request-headers", nil, false),
-								newTestCase(a[0], "/request-headers-notValues-bar", map[string]string{"x-foo": "foo"}, true),
-								newTestCase(a[0], "/request-headers-notValues-bar", map[string]string{"x-foo": "bar"}, false),
-
-								newTestCase(a[0], "/source-ip-a", nil, true),
-								newTestCase(b[0], "/source-ip-a", nil, false),
-								newTestCase(a[0], "/source-ip-b", nil, false),
-								newTestCase(b[0], "/source-ip-b", nil, true),
-								newTestCase(a[0], "/source-ip-notValues-b", nil, true),
-								newTestCase(b[0], "/source-ip-notValues-b", nil, false),
-
-								newTestCase(a[0], "/source-namespace-a", nil, true),
-								newTestCase(b[0], "/source-namespace-a", nil, false),
-								newTestCase(a[0], "/source-namespace-b", nil, false),
-								newTestCase(b[0], "/source-namespace-b", nil, true),
-								newTestCase(a[0], "/source-namespace-notValues-b", nil, true),
-								newTestCase(b[0], "/source-namespace-notValues-b", nil, false),
-
-								newTestCase(a[0], "/source-principal-a", nil, true),
-								newTestCase(b[0], "/source-principal-a", nil, false),
-								newTestCase(a[0], "/source-principal-b", nil, false),
-								newTestCase(b[0], "/source-principal-b", nil, true),
-								newTestCase(a[0], "/source-principal-notValues-b", nil, true),
-								newTestCase(b[0], "/source-principal-notValues-b", nil, false),
-
-								newTestCase(a[0], "/destination-ip-good", nil, true),
-								newTestCase(b[0], "/destination-ip-good", nil, true),
-								newTestCase(a[0], "/destination-ip-bad", nil, false),
-								newTestCase(b[0], "/destination-ip-bad", nil, false),
-								newTestCase(a[0], "/destination-ip-notValues-a-or-b", nil, true),
-								newTestCase(a[0], "/destination-ip-notValues-a-or-b-or-c", nil, false),
-
-								newTestCase(a[0], "/destination-port-good", nil, true),
-								newTestCase(b[0], "/destination-port-good", nil, true),
-								newTestCase(a[0], "/destination-port-bad", nil, false),
-								newTestCase(b[0], "/destination-port-bad", nil, false),
-								newTestCase(a[0], "/destination-port-notValues-c", nil, false),
-								newTestCase(b[0], "/destination-port-notValues-c", nil, false),
-
-								newTestCase(a[0], "/connection-sni-good", nil, true),
-								newTestCase(b[0], "/connection-sni-good", nil, true),
-								newTestCase(a[0], "/connection-sni-bad", nil, false),
-								newTestCase(b[0], "/connection-sni-bad", nil, false),
-								newTestCase(a[0], "/connection-sni-notValues-a-or-b", nil, true),
-								newTestCase(a[0], "/connection-sni-notValues-a-or-b-or-c", nil, false),
-
-								newTestCase(a[0], "/other", nil, false),
-								newTestCase(b[0], "/other", nil, false),
+								newTestCase(src1, "/request-headers", map[string]string{"x-foo": "foo"}, true),
```

Comment:
nit: you could probably enumerate these in a loop of `[]echo.Instances{src1, src2}` ?